File: chaincode/src/model/balancingDocument.ts

export class BalancingDocument {

    public docType: string;

    public balancingDocumentMrid?: string;
    public activationDocumentMrid?: string;
    public energyAmountMarketDocumentMrid?: string;
    public reserveBidMrid?: string;
    public revisionNumber: string;
    public messageType: string;
    public processsType: string;
    public senderMarketParticipantMrid?: string;
    public receiverMarketParticipantMrid?: string;
    public createdDateTime?: string;
    public period?: string;
    public businessType: string;
    public quantityMeasureUnitName?: string;
    public priceMeasureUnitName?: string;
    public currencyUnitName?: string;
    public meteringPointMrid?: string;
    public direction: string;
    public quantity?: number;
    public activationPriceAmount?: number;
    public financialPriceAmount?: number;
}

File: backend/service/src/main/java/com/star/models/balancing/BalancingDocument.java
package com.star.models.balancing;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * Copyright (c) 2022, Enedis (https://www.enedis.fr), RTE (http://www.rte-france.com)
 * SPDX-License-Identifier: Apache-2.0
 */
@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BalancingDocument {
    private String docType;
    private String balancingDocumentMrid;
    private String activationDocumentMrid;
    private String energyAmountMarketDocumentMrid;
    private String reserveBidMrid;
    private String revisionNumber;
    private String messageType;
    private String processsType;
    private String senderMarketParticipantMrid;
    private String receiverMarketParticipantMrid;
    private String createdDateTime;
    private String period;
    private String businessType;
    private String quantityMeasureUnitName;
    private String priceMeasureUnitName;
    private String currencyUnitName;
    private String meteringPointMrid;
    private String direction;
    private BigDecimal quantity;
    private BigDecimal activationPriceAmount;
    private BigDecimal financialPriceAmount;
}
